zilliqa/src/consensus.rs

```diff
@@ -485,10 +486,15 @@ impl Consensus {
                 break;
             }
 
-            // All TXs must have a corresponding account
-            if !self.state.has_account(tx.from_addr) {
+            // All TXs must have some balance
```

Comment:
Is this still a useful check? Presumably the transaction will fail anyway if the account has no balance (or insufficient balance).

What work do we save by failing early?

Reply:
Yup the tx will fail at the execution precheck. In all liklihood it doesn't change anything, but from a theoretical perspective you should always check the TX is valid otherwise someone else could fill your mempool with trash while getting the tx fees to themselves, essentially.